components/popup/test/popup.test.js

```diff
@@ -0,0 +1,96 @@
+import '../vwc-popup.js';
+import '@vonage/vwc-button';
+import 'chai-dom';
+import {
+	waitNextTask,
+	textToDomToParent,
+	isolatedElementsCreation
+} from '../../../test/test-helpers.js';
+import { chaiDomDiff } from '@open-wc/semantic-dom-diff';
+
+chai.use(chaiDomDiff);
+
+const COMPONENT_NAME = 'vwc-popup';
+
+describe('popup', () => {
+	let addElement = isolatedElementsCreation();
+
+	it(`${COMPONENT_NAME} is defined as a custom element`, async () => {
+		assert.exists(
+			customElements.get(COMPONENT_NAME)
+		);
+	});
+
+	it('should internal contents', async () => {
+		const addedElements = addElement(
+			textToDomToParent(`<${COMPONENT_NAME}></${COMPONENT_NAME}>`)
+		);
+		const actualElement = addedElements[0];
+		await waitNextTask();
+		expect(actualElement.shadowRoot.innerHTML).to.equalSnapshot();
+	});
+
+	describe('Popup default init', () => {
+		it('should reflect from attribute to property', async () => {
+			const [actualElement] = addElement(
+				textToDomToParent(`<${COMPONENT_NAME}></${COMPONENT_NAME}>`)
+			);
+
+			await actualElement.updateComplete;
+			expect(actualElement.anchor, 'anchor should be undefined')
+				.to
+				.equal(undefined);
+			expect(actualElement.open, 'open should be false')
+				.to
+				.equal(false);
+			expect(actualElement.arrow, 'arrow should be undefined')
+				.to
+				.equal(undefined);
+			expect(actualElement.corner, 'corner should be left')
+				.to
+				.equal('left');
+			expect(actualElement.dismissible, 'dismissible should be undefined')
+				.to
+				.equal(undefined);
+			expect(actualElement.distance, 'distance should be 10')
+				.to
+				.equal(10);
+			expect(actualElement.strategy, 'strategy should be absolute')
+				.to
+				.equal('absolute');
+		});
+	});
+
+	describe(`open`, function () {
+		it(`should be set to true`, function () {
+			const addedElements = addElement(
+				textToDomToParent(`<vwc-button></vwc-button>`)
+			);
+			const anchorElement = addedElements[0];
+			const [actualElement] = addElement(
+				textToDomToParent(`<${COMPONENT_NAME} open anchor=${anchorElement}></${COMPONENT_NAME}>`)
+			);
+
+			expect(actualElement.open)
```

Comment:
What happens if we don't have an anchor? Is there a test for that? I think I've added a comment about it in the `try/catch` clause

Reply:
@YonatanKra  I added tests without an anchor now